families/seth/rpc/src/account.rs

```diff
@@ -17,19 +17,111 @@
 
 use jsonrpc_core::{Params, Value, Error};
 
-use super::requests::{ValidatorClient};
+use super::requests::{
+    ValidatorClient,
+    BlockKey,
+    BlockKeyParseError,
+    num_to_hex,
+    bytes_to_hex,
+};
 
 use sawtooth_sdk::messaging::stream::MessageSender;
 use super::error;
 
-pub fn get_balance<T>(_params: Params, mut _client: ValidatorClient<T>) -> Result<Value, Error> where T: MessageSender {
-    Err(error::not_implemented())
+fn validate_block_key(block: String) -> Result<BlockKey, Error> {
+    match block.parse() {
+        Ok(k) => Ok(k),
+        Err(BlockKeyParseError::Invalid) => {
+            return Err(Error::invalid_params("Failed to parse block number"));
+        },
+        Err(BlockKeyParseError::Unsupported) => {
+            return Err(error::not_implemented());
+        },
+    }
 }
-pub fn get_storage_at<T>(_params: Params, mut _client: ValidatorClient<T>) -> Result<Value, Error> where T: MessageSender {
-    Err(error::not_implemented())
+
+fn validate_account_address(address: String) -> Result<String, Error> {
+    if address.len() != 42 {
+        Err(Error::invalid_params(
+            format!("Invalid address length: {} != {}", address.len(), 42)))
+    } else {
+        Ok(String::from(&address[2..]))
+    }
 }
-pub fn get_code<T>(_params: Params, mut _client: ValidatorClient<T>) -> Result<Value, Error> where T: MessageSender {
-    Err(error::not_implemented())
+
+fn validate_storage_address(address: String) -> Result<String, Error> {
+    if address.len() < 4 || address.len() % 2 != 0{
+        Err(Error::invalid_params(format!("Invalid storage position: {}", address)))
+    } else {
+        Ok(String::from(&address[2..]))
+    }
+}
+
+pub fn get_balance<T>(params: Params, mut client: ValidatorClient<T>) -> Result<Value, Error> where T: MessageSender {
+    info!("eth_getBalance");
+    let (address, block): (String, String) = match params.parse() {
```

Comment:
This might be doable more concisely as

```
let Ok(address, block): ParseResult<(String, String)> = params.parse().map_err(|_| Error::invalid_params("msg"))?;
```

Where in this case, 

```
type ParseResult<T> = Result<T, jsonrpc::Error>;
```

Would have to test...